#! /bin/sh /usr/share/dpatch/dpatch-run
## 07-ace-tao-ssliop-refcount.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix reference counter when using ssliop.
 
@DPATCH@
--- ACE_wrappers/TAO/orbsvcs/orbsvcs/SSLIOP/SSLIOP_Connector.cpp~	2007-01-25 10:04:11.000000000 -0800
+++ ACE_wrappers/TAO/orbsvcs/orbsvcs/SSLIOP/SSLIOP_Connector.cpp	2007-02-19 21:10:00.000000000 -0800
@@ -623,6 +623,11 @@
                                               remote_address,
                                               synch_options);
 
+      // base_connector_.connect() will increment the handler's #REFCOUNT#
+      // once more. This is not required as we already hold a reference to
+      // the handler, so we discard this second reference.
+      svc_handler->remove_reference ();
+
       // There are three possibilities from calling connect(): (a)
       // connection succeeds immediately - in this case, the
       // #REFCOUNT# on the handler is two; (b) connection completion
